File: 04_quicksort/swift/02_recursive_sum.swift
import Foundation

func sum(list : [Int]) -> Int {
    if list.count == 0 {
        return 0
    }
    var tempArray = list
    tempArray.remove(at: 0)
    return list[0] + sum(list: tempArray)
}

print(sum(list: [1, 2, 3, 4])) // => 10

File: 04_quicksort/golang/02_Recursive_sum.go
package main

import "fmt"

func sum(arr []int) int {
	if len(arr) == 0 {
		return 0
	}
	return arr[0] + sum(arr[1:])
}

func main() {
	fmt.Println(sum([]int{1, 2, 3, 4}))
}
